packages/cnpj-val/src/CnpjValidator.php

```
@@ -2,47 +2,163 @@
+            $sanitizedCnpj = strtoupper($sanitizedCnpj);
         }
 
-        $firstVerifierDigitIndex = CNPJ_LENGTH - 2;
-        $providedFirstVerifierDigit = $cnpjNumbersArray[$firstVerifierDigitIndex];
-        $baseFirstVerifierDigitCalculation = array_slice($cnpjNumbersArray, 0, $firstVerifierDigitIndex);
-        $calculatedFirstVerifierDigit = $this->verifierDigit->calculate($baseFirstVerifierDigitCalculation);
+        if ($actualOptions->type === CnpjValidationType::Numeric) {
+            $sanitizedCnpj = preg_replace('/[^0-9]/', '', $sanitizedCnpj) ?? '';
+        } else {
+            $sanitizedCnpj = preg_replace('/[^0-9A-Z]/i', '', $sanitizedCnpj) ?? '';
+        }
```

PHPMD (2.15.0)

[error] The method isValid uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)